client/state/terms/actions.js

```diff
@@ -16,14 +16,16 @@ import {
  * @param  {Number} siteId   Site ID
  * @param  {String} taxonomy Taxonomy Slug
  * @param  {Array}  terms    An array of term objects
+ * @param  {Object} query    Query Options
  * @return {Object}          Action object
  */
-export function receiveTerms( siteId, taxonomy, terms ) {
+export function receiveTerms( siteId, taxonomy, terms, query = {} ) {
 	return {
 		type: TERMS_RECEIVE,
 		siteId,
 		taxonomy,
-		terms
+		terms,
+		query
```

Comment:
Makes total sense.  The other route is to pass along the terms to `TERMS_REQUEST_SUCCESS` - would you be okay with that approach @aduth ?